### PR TITLE
Implemented dropdown for projects and work in progress on the mobile …

### PR DESCRIPTION
**Pull request by Hey-Zeus-byte**
…device NavBar
Implemented dropdown selection for "Projects" tab
Work in progress for the hamburger menu navbar
Screen Shot 2022-09-05 at 5 18 58 PM
Screen Shot 2022-09-05 at 5 19 20 PM

### CODE REVIEW

File: src/components/HamburgerButton.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import {useState} from "react";
+import styled from "styled-components";
+
+const HamburgerButtonWrapper = styled.div`
+  display: block;
+  margin: 1em;
```
Comment (fvaldez421) [inline src/components/HamburgerButton.js]: Why `em` here? Aren't we using `px`?

File: src/components/MobileNavLinksMenu.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import React, {useState} from "react";
+import {Link} from "react-router-dom";
+import styled from "styled-components";
+import {HamburgerButton} from "./HamburgerButton";
+
+const MobileNavLinkWrapper = styled.div`
+  @media only screen and (min-width: 768px) {
```
Comment (fvaldez421) [inline src/components/MobileNavLinksMenu.js:8]: this could be changed to just `display: none;`

File: src/components/MobileNavLinksMenu.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  @media only screen and (max-width: 768px) {
+    display: flex;
+    flex-flow: column;
+    background-color: rgb(245, 244, 220);
+    position: fixed;
+    top: 0;
+    right: 0;
+    height: 100vh;
+    width: 250px;
+    padding-top: 3.5rem;
```
Comment (fvaldez421) [inline src/components/MobileNavLinksMenu.js]: Why are we using `rem` here? You also used `em`... Lets stick to one.

File: src/components/HamburgerButton.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  div {
+    width: 30px;
+    height: 2px;
+    background-color: black;
+    margin: 8px 0;
+    border-radius: 5px;
+    transform-origin: 0;
+    transition: all 0.3s linear;
+    z-index: 20px;
```
Comment (fvaldez421) [inline src/components/HamburgerButton.js]: `z-index` doesn't use `px`. It works in levels starting at 0 as default

File: src/components/MobileNavLinksMenu.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import styled from "styled-components";
+import {HamburgerButton} from "./HamburgerButton";
+
+const MobileNavLinkWrapper = styled.div`
+  @media only screen and (min-width: 768px) {
+    display: none;
+  }
+  @media only screen and (max-width: 768px) {
+    display: flex;
+    flex-flow: column;
```
Comment (fvaldez421) [inline src/components/MobileNavLinksMenu.js]: What does `flex-flow` do here? Is this somehow different than `flex-direction`?

File: src/components/MobileNavLinksMenu.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  font-size: 11px;
+  line-height: 12px;
+  padding: 9px 20px;
+  font-weight: bold;
+
+  &:hover {
+    color: #000000;
+  }
+
+  @media (max-width: 768px) {
```
Comment (fvaldez421) [inline src/components/MobileNavLinksMenu.js]: ` only screen and `

File: src/components/MobileNavLinksMenu.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import {HamburgerButton} from "./HamburgerButton";
+
+const MobileNavLinkWrapper = styled.div`
+  @media only screen and (min-width: 768px) {
+    display: none;
+  }
+  @media only screen and (max-width: 768px) {
+    display: flex;
+    flex-flow: column;
+    background-color: rgb(245, 244, 220);
```
Comment (fvaldez421) [inline src/components/MobileNavLinksMenu.js]: Why are we using `rgb` here vs hex code? `#000000`

File: src/components/MobileNavLinksMenu.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+  text-decoration: none;
+  display: block;
+  text-align: left;
+  padding: 10px;
+`;
+
+const DropDownContent = styled.div`
+  display: none;
+  position: absolute;
+  background-color: white;
```
Comment (fvaldez421) [inline src/components/MobileNavLinksMenu.js]: lets switch this to hex code

File: src/components/NavBar.js [hunk 9]
```diff
@@ -28,29 +34,63 @@ const NavLink = styled(Link)`
...
   text-decoration: none;
   color: #407c51;
   font-size: 25px;
-  /* padding-left: 100px; */
 `;
 
+const DropDownContent = styled.div`
+  display: none;
+  position: absolute;
+  background-color: white;
```
Comment (fvaldez421) [inline src/components/NavBar.js:53]: lets switch this to hex code

File: src/components/HamburgerButton.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import styled from "styled-components";
+
+const HamburgerButtonWrapper = styled.div`
+  display: block;
+  margin: 1em;
+
+  div {
+    width: 30px;
+    height: 2px;
+    background-color: black;
```
Comment (fvaldez421) [inline src/components/HamburgerButton.js]: lets switch this to hex code

File: src/components/MobileNavLinksMenu.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    right: 0;
+    height: 100vh;
+    width: 250px;
+    padding-top: 3.5rem;
+  }
+`;
+
+const NavLink = styled(Link)`
+  text-decoration: none;
+  color: #919191;
```
Comment (fvaldez421) [inline src/components/MobileNavLinksMenu.js:29]: I would recommend we create a shared `theme` config that contains all these colors and their names/nicknames. That way we can import them across the project and be sure they're consistent.

File: src/components/HamburgerButton.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
+import {useState} from "react";
```
Comment (fvaldez421) [inline src/components/HamburgerButton.js]: didnt we talk about editing your prettier config file to use spaces between brackets? 
`{ useState }` vs `{useState}`

**Comment by Hey-Zeus-byte** [comment]
Commit: "Added new components and updated mobile NavBar" 
- Added a DropDownButton for the sub-menu(Mobile NavBar) links for John's projects. (Work in progress: Sub menu links do not open yet).
- Mobile NavBar opens when clicking on the hamburger menu button. Error on the homepage, but i believe the Mobile Navbar doesn't sit right due to the images on homepage. 
Screen Shot 2022-09-14 at 12 35 50 AM
Screen Shot 2022-09-14 at 12 35 58 AM

**Review by fvaldez421 — APPROVED** [review]

**Pull request merged**